services/fluxmonitor.go

```diff
@@ -354,9 +353,46 @@ func (fm *FluxMonitor) getAdapterResponse(endpoint url.URL, requestData string)
 	return response.Price, json.Unmarshal(body, &response)
 }
 
-func (fm *FluxMonitor) poll() error {
+// PollAndGetAnswer() will poll if the latest answer is older than the polling interval
+// and then return the latest value
+func (fm *FluxMonitor) PollAndGetAnswer() (decimal.Decimal, error) {
 	fm.pollMutex.Lock()
 	defer fm.pollMutex.Unlock()
+
+	if time.Since(fm.latestResultTimestamp)+fm.config.AdapterTimeout <= fm.config.PollInterval {
```

Comment:
This check can be moved to the only place that it is called, inside checkAndSendJob.

Reply:
Reasoning for the placement in the other comment